packages/modeldb-idb/src/ModelDB.ts

```diff
@@ -126,11 +132,13 @@ export class ModelDB extends AbstractModelDB {
 		return result as T[]
 	}
 
-	public async count(modelName: string): Promise<number> {
+	public async count(modelName: string, where?: WhereCondition): Promise<number> {
 		const api = this.#models[modelName]
 		assert(api !== undefined, `model ${modelName} not found`)
 		checkForMissingObjectStores(this.db, [api.storeName])
-		return await this.db.count(api.storeName)
+
+		const result = await this.read((txn) => api.count(txn, where), [api.storeName])
+		return result as number
```

Comment:
why do we have to cast `as number` here? can we just `return await this.read(...)`?